File: finance/src/main/java/mg/finance/entity/CurrencyConversion.java
package mg.finance.entity;

import lombok.*;

import javax.persistence.*;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Getter
@Setter
@Builder
@Table(name="mg_currency_conversion")
@NoArgsConstructor
@AllArgsConstructor
public class CurrencyConversion {

    @Id
    @GeneratedValue
    private UUID uuid;
    @ManyToOne
    @JoinColumn(name = "currency_from", nullable = false)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private Currency currencyFrom;
    @ManyToOne
    @JoinColumn(name = "currency_to", nullable = false)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private Currency currencyTo;
    private double value;
    private OffsetDateTime updatedOn;
}

File: webui/src/main/webapp/src/models/currency/currencyConversion.js
import {Currency} from "./currency";

export class CurrencyConversion {

    /**
     * @param {Object} config
     * @param {String} config.uuid
     * @param {Object} config.currencyFrom
     * @param {Object} config.currencyTo
     * @param {Number} config.value
     *
     */
    constructor(config) {
        config = config || {};
        this.uuid = config.uuid;
        this.currencyFrom = config.currencyFrom ? new Currency(config.currencyFrom) : null;
        this.currencyTo = config.currencyTo ? new Currency(config.currencyTo) : null;
        this.value = config.value;
    }
}